src/LPOracle.sol

```diff
@@ -1,7 +1,69 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.25;
 
+import { ICOWAMMPoolHelper } from "@cow-amm/interfaces/ICOWAMMPoolHelper.sol";
+import { GPv2Order } from "cowprotocol/contracts/libraries/GPv2Order.sol";
+import { IERC20 } from "cowprotocol/contracts/interfaces/IERC20.sol";
+
 contract LPOracle {
+    /// @notice BCoWPool address.
+    address public immutable POOL;
+    /// @notice BCoWHelper contract.
+    ICOWAMMPoolHelper public immutable HELPER;
+
+    /// @notice Pool tokens.
+    /// @dev Tokens at indicies 0 and 1 in `getFinalTokens` function. Use same order for input prices vector to `order`
+    /// function.
+    IERC20 public immutable TOKEN0;
+    IERC20 public immutable TOKEN1;
+
+    /// @notice Pool token 0 decimals
+    uint256 public immutable TOKEN0_DECIMALS;
+
+    /// @notice Pool token 1 decimals
+    uint256 public immutable TOKEN1_DECIMALS;
+
+    /// @dev Must check Chainlink price feeds match pool token ordering.
+    /// @param _pool BCoWPool address.
+    /// @param _helper BCoWHelper address.
+    constructor(address _pool, address _helper) {
+        /* Set pool and helper contracts */
+        POOL = _pool;
+        HELPER = ICOWAMMPoolHelper(_helper);
+
+        /* Gets pool tokens with correct ordering and pool validation checks */
+        address[] memory tokens = HELPER.tokens(POOL);
+        TOKEN0 = IERC20(tokens[0]);
+        TOKEN1 = IERC20(tokens[1]);
+
+        TOKEN0_DECIMALS = TOKEN0.decimals();
+        TOKEN1_DECIMALS = TOKEN1.decimals();
+    }
+
+    /// @notice Retrieves the order to satisfy the pool's invariants given the token prices.
+    /// @dev Zero fee constant function AMM.
+    /// @dev Decimals for price0 and price1 should be identical.
+    /// @param price0 USD price of pool token 0.
+    /// @param price1 USD price of pool token 1.
```

Comment:
Is this actually USD price? I suspect these are relative/intrinsic prices (independent of some external currency). 
```suggestion
    /// @param price0 price of pool token 0.
    /// @param price1 price of pool token 1.
```

Reply:
It isn't strictly enforced in this current implementation, however, we are heading in the direction where:
- prices obtained via chainlink USD price feeds (_getFeedData function, for example)
- these prices used as inputs to _simulateOrder.
- _simulateOrder then creates the input prices vector to the BCoWPool order function.

Reply:
Good to know. So they are USD prices and we can leave the docs strings like that. However, I guess `_normalizePrices` is agnostic to this fact and its docs strings can be left as they are (without it). 

I wonder if we have some other guarantees on the input from chainlink (cf the fuzzTest that checks within price bounds ranging within 10^18 apart).